pkg/controllers/status/cluster_status_controller.go

```diff
@@ -333,25 +364,20 @@ func generateReadyCondition(online, healthy bool) []metav1.Condition {
 	}
 
 	if !online {
-		conditions = append(conditions, newClusterOfflineCondition)
-	} else {
-		if !healthy {
-			conditions = append(conditions, newClusterNotReadyCondition)
-		} else {
-			conditions = append(conditions, newClusterReadyCondition)
-		}
+		return newClusterOfflineCondition
+	}
+	if !healthy {
+		return newClusterNotReadyCondition
 	}
 
-	return conditions
+	return newClusterReadyCondition
 }
 
-func setTransitionTime(oldClusterStatus, newClusterStatus *clusterv1alpha1.ClusterStatus) {
-	// preserve the last transition time if the status of member cluster not changed
-	if util.IsClusterReady(oldClusterStatus) == util.IsClusterReady(newClusterStatus) {
-		if len(oldClusterStatus.Conditions) != 0 {
-			for i := 0; i < len(newClusterStatus.Conditions); i++ {
-				newClusterStatus.Conditions[i].LastTransitionTime = oldClusterStatus.Conditions[0].LastTransitionTime
-			}
+func setTransitionTime(existingConditions []metav1.Condition, newCondition *metav1.Condition) {
```

Comment:
I think we don't need this function `setTransitionTime`. If we call `meta.SetStatusCondition`, the transition time will be automatically handled.

Reply:
> I think we don't need this function `setTransitionTime`. If we call `meta.SetStatusCondition`, the transition time will be automatically handled.

Thanks for your review, I will resolve the comment as soon as possible.
We need this function setTransitionTime because we always init a new `ClusterStatus` object, and the conditions are empty. 
```
var currentClusterStatus = clusterv1alpha1.ClusterStatus{}
```

Reply:
Got it.